pandas/tests/arrays/categorical/test_constructors.py

```diff
@@ -408,6 +408,11 @@ def test_constructor_str_unknown(self):
         with pytest.raises(ValueError, match="Unknown dtype"):
             Categorical([1, 2], dtype="foo")
 
+    def test_constructor_np_strs(self):
+        # GH#31499 Hastable.map_locations needs to work on np.str_ objects
+        cat = pd.Categorical(["1", "0", "1"], [np.str_("0"), np.str_("1")])
```

Comment:
actually, I think we should be santizing these inputs on Index construction; we already *do* this for Series IIRC (deep in block manager i think).  Can you see if that's better (as its a more maintainable long term soln). Not averse to your change, but that's *after* we already saved the inputs.